src/components/SlateEditor/plugins/table/index.tsx

```diff
@@ -268,55 +268,71 @@ export const tablePlugin = (editor: Editor) => {
-              const [, cellPath] = maybeNode;
-              const [parent] = Editor.node(editor, Path.parent(Path.parent(cellPath)));
-              const shouldHaveHeaders = !((node.rowHeaders && cellIndex === 0) || rowIndex === 0);
-              const headers = shouldHaveHeaders ? getHeader(matrix, rowIndex, cellIndex, node.rowHeaders) : undefined;
-
-              if (isTableHead(parent)) {
-                // If first row we add only a double digit id based on the cellIndex
-                if (rowIndex === 0 && cell.data.id !== `0${cellIndex}` && cell.type === TYPE_TABLE_CELL_HEADER) {
-                  return updateCell(editor, cell, { id: `0${cellIndex}` }, TYPE_TABLE_CELL_HEADER);
-                }
 
-                // Second head row need to have a id combined with the previous cell and headers are set to the standard ruleset
-                if (
-                  rowIndex === 1 &&
-                  matrix?.[1]?.[1]?.type === TYPE_TABLE_CELL_HEADER &&
-                  (cell.data.id !== `0${cellIndex}1${cellIndex}` ||
-                    (shouldHaveHeaders && headers !== cell.data.headers))
-                ) {
-                  return updateCell(
-                    editor,
-                    cell,
-                    { id: `0${cellIndex}1${cellIndex}`, headers: headers },
-                    TYPE_TABLE_CELL_HEADER,
-                  );
+      const tableHeadRows = Array.from(
+        editor.nodes({
+          match: (node, nodePath) => {
+            if (isTableRow(node)) {
+              const [parent] = Editor.parent(editor, nodePath);
+              return parent && isTableHead(parent);
+            }
+            return false;
+          },
+          at: path,
+        }),
+      );
+
+      const containsSpans = !!editor
+        .nodes({ match: (n) => isTableCell(n) && (n.data.colspan > 1 || n.data.rowspan > 1), at: path })
+        .next().value;
+
+      const containsIdOrHeaders = !!editor
+        .nodes({ match: (n) => isTableCell(n) && (!!n.data.id || !!n.data.headers), at: path })
+        .next().value;
+
+      const headerCellsInMultipleRows = tableHeadRows.length === 2 || (tableHeadRows.length >= 1 && node.rowHeaders);
+
+      // Should only have headers if a cell is associated with 2 or more header cells.
+      const shouldHaveHeaders = containsSpans || headerCellsInMultipleRows;
+      // Only remove headers if cell is only associated with 1 cell and there is cells with Id and headers in the table
+      const shouldRemoveHeaders = !shouldHaveHeaders && containsIdOrHeaders;
+
+      if (shouldHaveHeaders) {
+        return editor.withoutNormalizing(() => {
+          matrix?.forEach((row, rowIndex) => {
+            row.forEach((cell, cellIndex) => {
+              const [maybeNode] = Editor.nodes(editor, {
+                at: path,
+                match: (node) => isEqual(node, cell),
+              });
```

Comment:
Hvorfor trenger vi å hente ut noden her? Er ikke `cell` den samme noden?

Reply:
Hvis vi faktisk trenger å hente den ut: Kan vi gjøre det etter at vi har asserta at `previousMatrixCellIsEqualCurrent` treffer? Så gjør vi litt færre lookups.

Reply:
Mangler pathen, vi brukte `ReactEditor.findPath` før men den funka særs dårlig med endringene nå. Matrix lagrer vi kun elementet for en eller annen grunn 🇹🇭

Reply:
Mangler pathen, vi brukte `ReactEditor.findPath` før men den funka særs dårlig med endringene nå. Matrix lagrer vi kun elementet for en eller annen grunn 🇹🇭

Reply:
Mangler pathen, vi brukte `ReactEditor.findPath` før men den funka særs dårlig med endringene nå. Matrix lagrer vi kun elementet for en eller annen grunn 🇹🇭

Reply:
Kan vi endre matrix til å lagre node-entries?

Reply:
Jeg kan se på det, tror det blir mye endringer i matrixHelper